kernel/src/net/socket/netlink/route/message/segment/addr.rs

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::num::NonZeroU32;
+
+use align_ext::AlignExt;
+
+use super::{header::CMessageSegmentHeader, legacy::CRtGenMsg, NlSegmentCommonOps, SegmentBody};
+use crate::{
+    net::socket::netlink::route::message::{
+        attr::{addr::NlAddrAttr, NlAttr},
+        util::{align_reader, align_writer},
+        NLMSG_ALIGN,
+    },
+    prelude::*,
+    util::net::CSocketAddrFamily,
+};
+
+#[derive(Debug, Getters)]
+#[getset(get = "pub")]
+pub struct AddrSegment {
+    header: CMessageSegmentHeader,
+    body: AddrSegmentBody,
+    attrs: Vec<NlAddrAttr>,
+}
```

Comment:
Is it possible to create a generic structure so we will not need to repeat the `NlSegmentCommonOps` impls?
```rust
struct NlSegment<Body> {
    header: CMessageSegmentHeader,
    body: Body,
    attrs: Vec<NlAddrAttr>,
}
```